src/tests/ecdh_vectors.rs

```diff
@@ -0,0 +1,363 @@
+                    CKA_LABEL,
+                    format!(
+                        "{} public key, COUNT={}, line {}",
+                        unit.curve_name, unit.count, unit.line
+                    )
+                    .as_bytes()
+                )
+            ],
+            &[(CKA_DERIVE, true)],
+        ));
+
+        let derive_template = make_attr_template(
+            &[
+                (CKA_CLASS, CKO_SECRET_KEY),
+                (CKA_KEY_TYPE, CKK_GENERIC_SECRET),
+            ],
+            &[],
+            &[(CKA_EXTRACTABLE, true)],
+        );
+
+        let mut dk_handle = CK_INVALID_HANDLE;
+
+        let mut peer_point = test_to_ecc_point(&unit.cavs, unit.curve_name);
+
+        let mut params = CK_ECDH1_DERIVE_PARAMS {
+            kdf: CKD_NULL,
+            ulSharedDataLen: 0,
+            pSharedData: std::ptr::null_mut(),
+            ulPublicDataLen: peer_point.len() as CK_ULONG,
+            pPublicData: peer_point.as_mut_ptr(),
+        };
+        let mut mechanism: CK_MECHANISM = CK_MECHANISM {
+            mechanism: CKM_ECDH1_DERIVE,
+            pParameter: &mut params as *mut _ as CK_VOID_PTR,
+            ulParameterLen: std::mem::size_of::<CK_ECDH1_DERIVE_PARAMS>()
+                as CK_ULONG,
+        };
+
+        let ret = fn_derive_key(
+            session,
+            &mut mechanism,
+            priv_handle,
+            derive_template.as_ptr() as *mut _,
+            derive_template.len() as CK_ULONG,
+            &mut dk_handle,
+        );
+        if ret != CKR_OK {
+            if unit.fail {
+                continue;
+            }
+            panic!("Failed ({}) unit test at line {}", ret, unit.line);
+        }
+
+        let mut value = vec![0u8; unit.z.len()];
+        let mut extract_template = make_ptrs_template(&[(
+            CKA_VALUE,
+            void_ptr!(value.as_mut_ptr()),
+            value.len(),
+        )]);
+
+        let ret = fn_get_attribute_value(
+            session,
+            dk_handle,
+            extract_template.as_mut_ptr(),
+            extract_template.len() as CK_ULONG,
+        );
+        assert_eq!(ret, CKR_OK);
+
+        if unit.fail {
+            if value == unit.z {
+                // TODO This should be an error, but with 9, 10, 11, 12 (oi/tag/mac/dkm changed), the Z still
+                // matches and it fails here
+                if unit.errno < 9 {
+                    panic!("The unit test at line {} worked while it was expected to fail - values same [{}]", unit.line, hex::encode(value));
+                }
+            }
+        } else {
+            if value != unit.z {
+                panic!("Failed ({}) unit test at line {} - values differ [{} != {}]", ret, unit.line, hex::encode(value), hex::encode(unit.z));
+            }
+        }
+    }
+}
+
+#[test]
+#[parallel]
+fn test_ecdh_vector_init() {
+    let test_data = parse_ecdh_vector(
+        "testdata/KASValidityTest_ECCStaticUnified_KDFConcat_NOKC_init.fax",
```

Comment:
I don't think we need 2 sets of vetors, chose the one that is most representative (IIRC they just contain the same number and type of tests, the difference is one is the vector that was handed out and the other is an example of responses sent back for validation.

Usually the resp vector is the more useful, but often they have just identical type of data.